static/src/javascripts-legacy/projects/common/modules/atoms/youtube.js

```diff
@@ -184,6 +191,8 @@ define([
 
                 // append index of atom as atomId must be unique
                 var atomId = el.getAttribute('data-media-atom-id') + '/' + index;
+                //need data attribute with index for unique lookup
+                el.setAttribute('data-unique-atom-id', atomId);
```

Comment:
This contains the client side computed index number so has to be done client side.